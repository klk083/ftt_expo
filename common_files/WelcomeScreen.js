import React, {Component} from 'react';
import {StyleSheet, Text, View, Image, Button, TouchableOpacity, SafeAreaView} from 'react-native';
import { RFPercentage } from "react-native-responsive-fontsize";

import { privacyTitle, terms_of_serviceTitle, welcome_to, read_our, confirmPrivacyTermsOfService, acceptAndContinue} from "./Texts";

export default class WelcomeScreen extends Component {
  render() {
    return (
        <SafeAreaView style={styles.safeAreaContainer}>
          <View style={styles.container}>
            <View style={styles.welcomeTextContainer}>
              <Text style={styles.welcomeText}>{welcome_to}</Text>
            </View>
            <Image
                source={require('../assets/fast_track_taxi_logo_ferdig.png')}
                style={styles.logoContainer}
            />
            <View style={styles.infoContainer}>
              <View style={styles.infoTextContainer}>
              <Text style={styles.instructions}>{read_our}
              <Text style={styles.linked} onPress={() => this.props.navigation.navigate('Privacy')}
              >{privacyTitle}</Text>{confirmPrivacyTermsOfService}
                <Text style={styles.linked}
                      onPress={() => this.props.navigation.navigate('TermsOfService')}
                >{terms_of_serviceTitle}</Text>.
              </Text>
              </View>
              <View style={styles.buttonContainer}>
                <TouchableOpacity style={styles.touchableButtonContainer}>
                  <Text style={styles.confirmButton} onPress={() => {
                    this.props.navigation.navigate('Number_registration')
                  }}>{acceptAndContinue}</Text>
                </TouchableOpacity>
              </View>
            </View>
          </View>
        </SafeAreaView>
    );
  }
}

const styles = StyleSheet.create({
  safeAreaContainer: {
    flex: 1,
  },
  container: {
    flex: 1,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
  },
  welcomeTextContainer: {
    flex: 1,
    alignItems: 'center',
    alignContent: 'center',
    justifyContent: 'center',
  },
  logoContainer: {
    flex: 2,
    width: 380,
    height: 380,
  },
  infoContainer: {
    flex: 1,
    justifyContent: 'flex-end',
    padding: 20,
  },
  welcomeText: {
    fontSize: RFPercentage(8),
    textAlign: 'center',
    textAlignVertical: 'center',
    alignItems: 'stretch',
  },
  instructions: {
    flex: 1,
    //marginHorizontal: 30,
    //lineHeight: 30,
    textAlign: 'center',
    //color: '#000000',
    fontSize: RFPercentage(3),
    //marginBottom: 5,
  },
  buttonContainer: {
    flex: 1,
    justifyContent: 'center',
    //margin: 20,
    //padding: 15,
  },
  linked: {
    //marginTop: 10,
    textAlign: 'center',
    //margin: 10,
    color: 'blue',
    textDecorationLine: 'underline',
  },


  infoTextContainer: {
    flex: 1,
    justifyContent: 'space-evenly',
  },
  touchableButtonContainer: {
    flex: 1,
    justifyContent: 'space-evenly',
    alignItems: 'center',
    borderRadius: 25,
    backgroundColor: 'darkseagreen',
  },
  confirmButton: {
    textTransform: 'uppercase',
    alignItems: 'center',
    fontSize: RFPercentage(4),
  },
});
