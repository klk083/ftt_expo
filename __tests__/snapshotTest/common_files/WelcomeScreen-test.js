import React from 'react'
import WelcomeScreen from '../../../common_files/WelcomeScreen'

import renderer from 'react-test-renderer'

test('renders correctly', () => {
    const tree = renderer.create(<WelcomeScreen />).toJSON()
    expect(tree).toMatchSnapshot()
})
