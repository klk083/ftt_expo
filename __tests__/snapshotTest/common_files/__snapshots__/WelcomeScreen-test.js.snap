// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "flex": 1,
        "justifyContent": "space-evenly",
      }
    }
  >
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "alignItems": "stretch",
            "fontSize": 107,
            "textAlign": "center",
            "textAlignVertical": "center",
          }
        }
      >
        Velkommen til
      </Text>
    </View>
    <Image
      source={
        Object {
          "testUri": "../../../assets/fast_track_taxi_logo_ferdig.png",
        }
      }
      style={
        Object {
          "flex": 2,
          "height": 380,
          "width": 380,
        }
      }
    />
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "flex-end",
          "padding": 20,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "space-evenly",
          }
        }
      >
        <Text
          style={
            Object {
              "flex": 1,
              "fontSize": 40,
              "textAlign": "center",
            }
          }
        >
          Les vår 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
                "textAlign": "center",
                "textDecorationLine": "underline",
              }
            }
          >
            Personvern
          </Text>
          . Trykk "Bekreft og fortsett" for å akseptere 
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "blue",
                "textAlign": "center",
                "textDecorationLine": "underline",
              }
            }
          >
            Servicevilkår
          </Text>
          .
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <View
          accessible={true}
          focusable={false}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "darkseagreen",
              "borderRadius": 25,
              "flex": 1,
              "justifyContent": "space-evenly",
              "opacity": 1,
            }
          }
        >
          <Text
            onPress={[Function]}
            style={
              Object {
                "alignItems": "center",
                "fontSize": 53,
                "textTransform": "uppercase",
              }
            }
          >
            Bekreft og fortsett
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
